pkg/util/signals/signals.go

```diff
@@ -18,12 +18,28 @@ package signals
 
 import (
 	"context"
+	"os"
 	"os/signal"
 	"syscall"
+	"time"
 )
 
 // NewSigKillContext returns a Context that cancels when os.Interrupt or os.Kill is received
 func NewSigKillContext() context.Context {
 	ctx, _ := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
 	return ctx
 }
+
+// NewSigTermHandler creates a channel to listen to SIGTERM and runs the handle function
+func NewSigTermHandler(ctx context.Context, handle func()) {
+	_, cancel := context.WithCancel(ctx)
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGTERM)
+
+	go func() {
+		<-c
+		handle()
+		time.Sleep(30 * time.Second)
```

Comment:
Oh, I see what you did. I would take the logic out of here - I don't think this function even needs to be context aware. I would especially take the sleep out of here, it's pretty bespoke.

Reply:
Yea your suggestion looks and feels much cleaner.